kratos/python_scripts/read_materials_process.py

```diff
@@ -108,7 +108,18 @@ def _GetConstitutiveLaw(self, my_string):
 
         model_part.GetProperties(prop_id).SetValue(CONSTITUTIVE_LAW, constitutive_law)
         """
-        return self._get_attribute(my_string, KratosMultiphysics.KratosGlobals.GetConstitutiveLaw, "Constitutive Law")
+        my_string = param["name"].GetString()
+        splitted = my_string.split(".")
+
+        if len(splitted) == 0:
+            raise Exception("Something wrong. Trying to split the string " + my_string)
+        if len(splitted) > 3:
+            raise Exception("Something wrong. String " + my_string + " has too many arguments")
+
+        cl_name = splitted[-1]
+        param["name"].SetString(cl_name)
+        cl = self._get_attribute(cl_name, KratosMultiphysics.KratosGlobals.GetConstitutiveLaw, "Constitutive Law")
+        return cl.Create(param)
```

Comment:
For this to work properly the Create function has to either be implemented in every CL or (what you are using) by passing the name of the CLaw within the Parameters
=> [this Function is virtual](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/constitutive_law.h#L518), and I don't see why
I.e. if someone overrides this function and does not pass the name, the proposed mechanism will fail!
Bottom line: Is this virtual really needed?

Reply:
If the Create is not defined in every CL will take the one from the base class, so I will clone a CL from the KratosComponents, the same behaviour we were using until now